### PR TITLE
[KnownAsImmutable] -> [TreatAsImmutable]

### PR DESCRIPTION
**Pull request by carldebilly**
The working is better that way
+ better error messages.

Another fix for #52

### CODE REVIEW

File: doc/Immutable Generation.md [hunk 1]
```diff
@@ -367,11 +367,11 @@ public class MyImmutable
 I'm getting this error:
 
 ``` csharp
-#error: 'ImmutableGenerator: Property MyClass.SomeField type ExternalClass.SuperClass is not immutable. It cannot be used in an immutable entity.'
+#error: 'ImmutableGenerator: Property MyClass.SomeField (ClassBase) is not immutable. It cannot be used in an immutable entity.'
```
Comment (dr1rrb) [inline doc/Immutable Generation.md]: BaseClass ?

File: src/Uno.CodeGen/ImmutableGenerator.cs [hunk 2]
```diff
@@ -859,22 +861,24 @@ void CheckTypeImmutable(ITypeSymbol type, string typeSource, bool constraintsChe
 				if (type.FindAttribute(_immutableBuilderAttributeSymbol) != null)
 				{
 					builder.AppendLineInvariant(
-						$"#error {nameof(ImmutableGenerator)}: {typeSource} type {type} IS A BUILDER! It cannot be used in an immutable entity.");
+						$"#error {nameof(ImmutableGenerator)}: {typeSource} ({typeName}) IS A BUILDER! It cannot be used in an immutable entity.");
```
Comment (jeromelaban) [inline src/Uno.CodeGen/ImmutableGenerator.cs]: there is no need for all caps :)

File: doc/Immutable Generation.md (hunk 1 above)
Comment (carldebilly) [reply doc/Immutable Generation.md]: Good catch.  J'ai tout changé sauf ça.

File: src/Uno.CodeGen/ImmutableGenerator.cs (hunk 2 above)
Comment (carldebilly) [reply src/Uno.CodeGen/ImmutableGenerator.cs]: I feel like I'm menacing an ennemi country. Maybe it's too much ;-).

File: doc/Immutable Generation.md (hunk 1 above)
Comment (carldebilly) [reply doc/Immutable Generation.md]: Ohhh that's just the name you don't like, it's not a typo from me.  Ok, I'll change it anyway.

**Pull request merged**